jetty-ee10/jetty-ee10-servlets/src/main/java/org/eclipse/jetty/ee10/servlets/EventSourceServlet.java

```diff
@@ -199,17 +200,20 @@ public void run()
                 // We could write, reschedule heartbeat
                 scheduleHeartBeat();
             }
-            catch (IOException x)
+            catch (Throwable x)
             {
+                IO.close(this::close);
+
                 try
                 {
                     // The other peer closed the connection
-                    close();
                     eventSource.onClose();
                 }
                 catch (Throwable t)
                 {
-                    t.printStackTrace();
+                    if (t != x)
```

Comment:
This should have been `ExceptionUtil.addSuppressedIfNotAssociated(x, t)`

Reply:
Thanks.  I'll adjust HEAD